extractor.py

```diff
@@ -1,146 +1,179 @@
+
+def _guess_type(filename):
+    return mimetypes.guess_type(filename, False)
+
+def extract(filename):
+    print("Will try to extract file:", filename)
+    ext, encoding = _guess_type(filename)
+    if not ext and not encoding:
+        print("Could not guess filetype. Cannot extract")
+        return None
+    print(ext)
+    config_result = _check_extension(ext, encoding)
+    logger.logd('ExtractorConfig:', config_result)
+    if config_result[_error_str]:
+        print("Cannot extract at this moment")
+        return None
+
+    result = config_result[_result_str]
+    if not result[_result_str]:
+        # Need to create file
+        logger.logd("create file for configurations")
+        _create_configuration_file()
+
+    prog_config = result[_prog_str]
+    if not prog_config:
+        # Add configuration for this extension
+        logger.logd('add config for this extension', ext, encoding)
+        print('We are going to need to add a new config for', filename)
+        prog_config = _configure_new_item(ext, encoding)
+        if not prog_config:
+            print('Could not extract for this type of file at this moment.')
+            logger.logd('User did not enter a new configuration')
+            return
 
-    def build_command(self, program, filename):
-        if not program:
-            return None
-        if not filename:
-            return None
+    logger.logd('Found config:', prog_config)
 
-    def guess_type(self, filename):
-        ext = mimetypes.guess_type(filename, False)
-        if not ext:
-            return None
-        if DEBUG:
-            print(ext)
-        return ext
+    try:
+        prog_command = prog_config[_prog_str]
+    except:
+        print('Could not extract')
+        return None
 
-    def extract(self, filename):
-        print("Will try to extract file:", filename)
-        ext = self.guess_type(filename)
-        if not ext:
-            print("Could not guess filetype. Cannot extract")
-            return None
-        prog = Program()
-        _ext = prog.get_proper_ext(ext)
-        if not _ext:
-            print("Cannot extract at this moment")
-            return None
+    prog = Program(prog_command)
+    prog.execute(prog_config[_attrs_str], filename)
 
-        _prog = prog.get_prog(_ext)
+class Program:
+    prog = None
 
-        _command = prog.get_full_command(_prog, _ext, filename)
-        if DEBUG:
-            print(_command)
-        if not _command:
-            return
+    def __init__(self, prog):
+        self.prog = prog
 
-        os.system(_command)
+    def execute(self, attrs, filename):
+        if not filename:
+            print("Could not get the name of your file")
+            return None
+        if not attrs:
+            attrs = ''
+        full_command = self.prog + ' ' + attrs + ' ' + filename
+        logger.logd('execute:', full_command)
+        os.system(full_command)
 
 def main():
```

Comment:
maybe we should name it `_test()` to make clear this method is not part of normal program flow ?

also note the underscore.